app/assets/javascripts/govuk_publishing_components/components/cookie-banner.js

```diff
@@ -9,18 +9,26 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
     this.$module.hideCookieMessage = this.hideCookieMessage.bind(this)
     this.$module.showConfirmationMessage = this.showConfirmationMessage.bind(this)
     this.$module.setCookieConsent = this.setCookieConsent.bind(this)
+    this.$module.currentPage = window.location.pathname
 
-    this.$hideLink = this.$module.querySelector('a[data-hide-cookie-banner], button[data-hide-cookie-banner]')
-    if (this.$hideLink) {
-      this.$hideLink.addEventListener('click', this.$module.hideCookieMessage)
-    }
+    var newCookieBanner = document.querySelector('.gem-c-cookie-banner--new')
 
-    this.$acceptCookiesLink = this.$module.querySelector('button[data-accept-cookies]')
-    if (this.$acceptCookiesLink) {
-      this.$acceptCookiesLink.addEventListener('click', this.$module.setCookieConsent)
-    }
+    // Hide the cookie banner on the cookie settings page, to avoid circular journeys
+    if (newCookieBanner && this.$module.currentPage === '/help/cookies') {
```

Comment:
I wouldn't bother setting a property `this.$module.currentPage` if it's only used once, think the comment on line 16 is good enough to show the intent 👍 

```suggestion
    if (newCookieBanner && window.location.pathname === '/help/cookies') {
```